File: hw/chavez/u3/hw28Strategy/Java/Strategy/src/main/java/ec/edu/espe/strategy/controller/SortingContext.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.strategy.controller;

import ec.edu.espe.strategy.model.BubbleSort;
import ec.edu.espe.strategy.model.InsertionSort;
import ec.edu.espe.strategy.model.QuickSort;
import ec.edu.espe.strategy.model.SortingStrategy;



/**
 *
 * @author Odalys Chavez, CodeBreakers, @ESPE
 */
public class SortingContext {
    private SortingStrategy strategy;

    public int[] sort(int[] data) {

        if (data.length < 30) {
            strategy = new BubbleSort();
        } else if (data.length < 100) {
            strategy = new InsertionSort();
        } else {
            strategy = new QuickSort();
        }

        return strategy.sort(data);
    }
    
}

File: hw/chavez/u3/hw28Strategy/Python/controller/sorting_context.py
from model.bubble_sort import BubbleSort
from model.insertion_sort import InsertionSort
from model.quick_sort import QuickSort
class SortingContext:

    def sort(self, data):

        if len(data) < 30:
            strategy = BubbleSort()
        elif len(data) < 100:
            strategy = InsertionSort()
        else:
            strategy = QuickSort()

        return strategy.sort(data)